ConsentViewController/Classes/SPConsentManager.swift

```diff
@@ -497,9 +497,7 @@ extension SPConsentManager: SPMessageUIDelegate {
     }
 
     func onAction(_ action: SPAction, from controller: UIViewController) {
-        DispatchQueue.main.async { [weak self] in
```

Comment:
@andresilveirah  why is it not replaced with "DispatchQueue.main.sync"? Just curious.

Reply:
@Nevazhnovu that's a good question. I first tried using `main.sync` but it turns out if the process is being executed in the main thread, running `main.sync` throws a run time exception. The solution is to simply run the code you want without dispatching it to the main queue.

Reply:
@andresilveirah ahh sweet sweet deadlock. Got it